### ISSUE TITLE
CI: ConformanceKind1.19: connectivity test failed: 5 tests failed: Connectivity to 1.1.1.1 is broken

### ISSUE THREAD

**Issue by joestringer** [cilium/cilium#22217]
The ConformanceKind workflow seems to be unstable on master while connecting to 1.1.1.1.

Example run:
https://github.com/cilium/cilium/actions/runs/3468759658/jobs/5794941960
[cilium-sysdump-out.zip (28).zip](https://github.com/cilium/cilium/files/10024188/cilium-sysdump-out.zip.28.zip)

```
Run cilium connectivity test --flow-validation=disabled --hubble=false
  cilium connectivity test --flow-validation=disabled --hubble=false
  shell: /usr/bin/bash -e {0}
  env:
    kind_version: v0.11.1
    kind_config: .github/kind-config.yaml
    cilium_cli_version: v0.1[2](https://github.com/cilium/cilium/actions/runs/3468759658/jobs/5794941960#step:12:2).4
    QUAY_ORGANIZATION: cilium
    QUAY_ORGANIZATION_DEV: cilium
✨ [kind-chart-testing] Creating namespace cilium-test for connectivity check...
✨ [kind-chart-testing] Deploying echo-same-node service...
✨ [kind-chart-testing] Deploying DNS test server configmap...
✨ [kind-chart-testing] Deploying same-node deployment...
✨ [kind-chart-testing] Deploying client deployment...
✨ [kind-chart-testing] Deploying client2 deployment...
✨ [kind-chart-testing] Deploying echo-other-node service...
✨ [kind-chart-testing] Deploying other-node deployment...
⌛ [kind-chart-testing] Waiting for deployments [client client2 echo-same-node] to become ready...
⌛ [kind-chart-testing] Waiting for deployments [echo-other-node] to become ready...
⌛ [kind-chart-testing] Waiting for CiliumEndpoint for pod cilium-test/client-7bdbddd7b-bc67l to appear...
⌛ [kind-chart-testing] Waiting for CiliumEndpoint for pod cilium-test/client2-6ff4656b7d-wjj8j to appear...
⌛ [kind-chart-testing] Waiting for pod cilium-test/client-7bdbddd7b-bc67l to reach DNS server on cilium-test/echo-same-node-77494b4cbd-shdlv pod...
⌛ [kind-chart-testing] Waiting for pod cilium-test/client2-6ff4656b7d-wjj8j to reach DNS server on cilium-test/echo-same-node-77494b4cbd-shdlv pod...
⌛ [kind-chart-testing] Waiting for pod cilium-test/client-7bdbddd7b-bc67l to reach DNS server on cilium-test/echo-other-node-c7cd7697d-sqdsb pod...
⌛ [kind-chart-testing] Waiting for pod cilium-test/client2-6ff4656b7d-wjj8j to reach DNS server on cilium-test/echo-other-node-c7cd7697d-sqdsb pod...
⌛ [kind-chart-testing] Waiting for pod cilium-test/client-7bdbddd7b-bc67l to reach default/kubernetes service...
⌛ [kind-chart-testing] Waiting for pod cilium-test/client2-6ff4656b7d-wjj8j to reach default/kubernetes service...
⌛ [kind-chart-testing] Waiting for CiliumEndpoint for pod cilium-test/echo-other-node-c7cd7697d-sqdsb to appear...
⌛ [kind-chart-testing] Waiting for CiliumEndpoint for pod cilium-test/echo-same-node-77494b4cbd-shdlv to appear...
⌛ [kind-chart-testing] Waiting for Service cilium-test/echo-other-node to become ready...
⌛ [kind-chart-testing] Waiting for Service cilium-test/echo-same-node to become ready...
⌛ [kind-chart-testing] Waiting for NodePort 172.18.0.2:[3](https://github.com/cilium/cilium/actions/runs/3468759658/jobs/5794941960#step:12:3)2177 (cilium-test/echo-other-node) to become ready...
⌛ [kind-chart-testing] Waiting for NodePort 172.18.0.2:30292 (cilium-test/echo-same-node) to become ready...
⌛ [kind-chart-testing] Waiting for NodePort 172.18.0.3:32177 (cilium-test/echo-other-node) to become ready...
⌛ [kind-chart-testing] Waiting for NodePort 172.18.0.3:30292 (cilium-test/echo-same-node) to become ready...
ℹ️  Skipping IPCache check
ℹ️  Cilium version: 1.12.90
🏃 Running tests...

[=] Test [no-policies]
...................................
  [-] Scenario [no-policies/pod-to-pod]
  [.] Action [no-policies/pod-to-pod/curl-0: cilium-test/client2-6ff[4](https://github.com/cilium/cilium/actions/runs/3468759658/jobs/5794941960#step:12:4)6[5](https://github.com/cilium/cilium/actions/runs/3468759658/jobs/5794941960#step:12:5)[6](https://github.com/cilium/cilium/actions/runs/3468759658/jobs/5794941960#step:12:6)b[7](https://github.com/cilium/cilium/actions/runs/3468759658/jobs/5794941960#step:12:7)d-wjj[8](https://github.com/cilium/cilium/actions/runs/3468759658/jobs/5794941960#step:12:8)j (10.244.1.237) -> cilium-test/echo-other-node-c7cd76[9](https://github.com/cilium/cilium/actions/runs/3468759658/jobs/5794941960#step:12:9)7d-sqdsb ([10](https://github.com/cilium/cilium/actions/runs/3468759658/jobs/5794941960#step:12:11).244.0.57:8080)]
  [.] Action [no-policies/pod-to-pod/curl-1: cilium-test/client2-6ff4656b7d-wjj8j (10.244.1.237) -> cilium-test/echo-same-node-77494b4cbd-shdlv (10.244.1.233:8080)]
  [.] Action [no-policies/pod-to-pod/curl-2: cilium-test/client-7bdbddd7b-bc67l (10.244.1.97) -> cilium-test/echo-other-node-c7cd7697d-sqdsb (10.244.0.57:8080)]
  [.] Action [no-policies/pod-to-pod/curl-3: cilium-test/client-7bdbddd7b-bc67l (10.244.1.97) -> cilium-test/echo-same-node-77494b4cbd-shdlv (10.244.1.233:8080)]
  [-] Scenario [no-policies/pod-to-hostport]
  [.] Action [no-policies/pod-to-hostport/curl-0: cilium-test/client-7bdbddd7b-bc67l (10.244.1.97) -> cilium-test/echo-same-node-77494b4cbd-shdlv (172.18.0.3:40000)]
  [.] Action [no-policies/pod-to-hostport/curl-1: cilium-test/client-7bdbddd7b-bc67l (10.244.1.97) -> cilium-test/echo-other-node-c7cd7697d-sqdsb (172.18.0.2:40000)]
  [-] Scenario [client-egress-to-cidr-deny/pod-to-cidr]
  [.] Action [client-egress-to-cidr-deny/pod-to-cidr/cloudflare-1001-0: cilium-test/client-7bdbddd7b-bc67l (10.244.1.97) -> cloudflare-1001 (1.0.0.1:80)]
  [.] Action [client-egress-to-cidr-deny/pod-to-cidr/cloudflare-1001-1: cilium-test/client2-6ff4656b7d-wjj8j (10.244.1.237) -> cloudflare-1001 (1.0.0.1:80)]
  [.] Action [client-egress-to-cidr-deny/pod-to-cidr/cloudflare-[11](https://github.com/cilium/cilium/actions/runs/3468759658/jobs/5794941960#step:12:12)11-0: cilium-test/client-7bdbddd7b-bc67l (10.244.1.97) -> cloudflare-1111 (1.1.1.1:80)]
  ❌ command "curl -w %{local_ip}:%{local_port} -> %{remote_ip}:%{remote_port} = %{response_code} --silent --fail --show-error --connect-timeout 5 --output /dev/null http://1.1.1.1:80" failed: command terminated with exit code 28
  ℹ️  curl output:
  curl: (28) Connection timeout after 5000 ms
:0 -> :0 = 000
  
  📄 No flows recorded during action cloudflare-1111-0
  📄 No flows recorded during action cloudflare-1111-0
  [.] Action [client-egress-to-cidr-deny/pod-to-cidr/cloudflare-1111-1: cilium-test/client2-6ff4656b7d-wjj8j (10.244.1.237) -> cloudflare-1111 (1.1.1.1:80)]
  ❌ command "curl -w %{local_ip}:%{local_port} -> %{remote_ip}:%{remote_port} = %{response_code} --silent --fail --show-error --connect-timeout 5 --output /dev/null http://1.1.1.1:80" failed: command terminated with exit code 28
  ℹ️  curl output:
  curl: (28) Connection timeout after 5000 ms
:0 -> :0 = 000
  
  📄 No flows recorded during action cloudflare-1111-1
  📄 No flows recorded during action cloudflare-1111-1
  ℹ️  📜 Deleting CiliumNetworkPolicy 'allow-all-egress' from namespace 'cilium-test'..
  ℹ️  📜 Deleting CiliumNetworkPolicy 'client-egress-to-cidr-deny' from namespace 'cilium-test'..

[=] Test [health]
..
[=] Test [client-egress-l7-method]
............
[=] Test [client-egress-l7]
..........
[=] Test [client-egress-l7-named-port]
........
  ℹ️  📜 Applying CiliumNetworkPolicy 'client-egress-only-dns' to namespace 'cilium-test'..
  ℹ️  📜 Applying CiliumNetworkPolicy 'client-egress-l7-http-named-port' to namespace 'cilium-test'..
  [-] Scenario [client-egress-l7-named-port/pod-to-pod]
  [.] Action [client-egress-l7-named-port/pod-to-pod/curl-0: cilium-test/client-7bdbddd7b-bc67l (10.244.1.97) -> cilium-test/echo-other-node-c7cd7697d-sqdsb (10.244.0.57:8080)]
  [.] Action [client-egress-l7-named-port/pod-to-pod/curl-1: cilium-test/client-7bdbddd7b-bc67l (10.244.1.97) -> cilium-test/echo-same-node-77494b4cbd-shdlv (10.244.1.233:8080)]
  [.] Action [client-egress-l7-named-port/pod-to-pod/curl-2: cilium-test/client2-6ff4656b7d-wjj8j (10.244.1.237) -> cilium-test/echo-other-node-c7cd7697d-sqdsb (10.244.0.57:8080)]
  [.] Action [client-egress-l7-named-port/pod-to-pod/curl-3: cilium-test/client2-6ff4656b7d-wjj8j (10.244.1.237) -> cilium-test/echo-same-node-77494b4cbd-shdlv (10.244.1.233:8080)]
  [-] Scenario [client-egress-l7-named-port/pod-to-world]
  [.] Action [client-egress-l7-named-port/pod-to-world/http-to-one-one-one-one-0: cilium-test/client-7bdbddd7b-bc67l (10.244.1.97) -> one-one-one-one-http (one.one.one.one:80)]
  [.] Action [client-egress-l7-named-port/pod-to-world/https-to-one-one-one-one-0: cilium-test/client-7bdbddd7b-bc67l (10.244.1.97) -> one-one-one-one-https (one.one.one.one:443)]
  [.] Action [client-egress-l7-named-port/pod-to-world/https-to-one-one-one-one-index-0: cilium-test/client-7bdbddd7b-bc67l (10.244.1.97) -> one-one-one-one-https-index (one.one.one.one:443)]
  [.] Action [client-egress-l7-named-port/pod-to-world/http-to-one-one-one-one-1: cilium-test/client2-6ff4656b7d-wjj8j (10.244.1.237) -> one-one-one-one-http (one.one.one.one:80)]
  ❌ command "curl -w %{local_ip}:%{local_port} -> %{remote_ip}:%{remote_port} = %{response_code} --silent --fail --show-error --connect-timeout 5 --output /dev/null http://one.one.one.one:80" failed: command terminated with exit code [22](https://github.com/cilium/cilium/actions/runs/3468759658/jobs/5794941960#step:12:23)
  ℹ️  curl output:
  curl: (22) The requested URL returned error: 503
10.244.1.[23](https://github.com/cilium/cilium/actions/runs/3468759658/jobs/5794941960#step:12:24)7:52666 -> 1.1.1.1:80 = 503
  
  📄 No flows recorded during action http-to-one-one-one-one-1
  📄 No flows recorded during action http-to-one-one-one-one-1
  [.] Action [client-egress-l7-named-port/pod-to-world/https-to-one-one-one-one-1: cilium-test/client2-6ff4656b7d-wjj8j (10.[24](https://github.com/cilium/cilium/actions/runs/3468759658/jobs/5794941960#step:12:25)4.1.237) -> one-one-one-one-https (one.one.one.one:443)]
  [.] Action [client-egress-l7-named-port/pod-to-world/https-to-one-one-one-one-index-1: cilium-test/client2-6ff4656b7d-wjj8j (10.244.1.237) -> one-one-one-one-https-index (one.one.one.one:443)]
  ℹ️  📜 Deleting CiliumNetworkPolicy 'client-egress-only-dns' from namespace 'cilium-test'..
  ℹ️  📜 Deleting CiliumNetworkPolicy 'client-egress-l7-http-named-port' from namespace 'cilium-test'..

[=] Test [dns-only]
..........
[=] Test [to-fqdns]
........
📋 Test Report
connectivity test failed: 5 tests failed
❌ 5/30 tests failed (9/2[28](https://github.com/cilium/cilium/actions/runs/3468759658/jobs/5794941960#step:12:29) actions), 0 tests skipped, 0 scenarios skipped:
Test [no-policies]:
  ❌ no-policies/pod-to-cidr/cloudflare-1111-0: cilium-test/client-7bdbddd7b-bc67l (10.244.1.97) -> cloudflare-1111 (1.1.1.1:80)
  ❌ no-policies/pod-to-cidr/cloudflare-1111-1: cilium-test/client2-6ff4656b7d-wjj8j (10.244.1.2[37](https://github.com/cilium/cilium/actions/runs/3468759658/jobs/5794941960#step:12:38)) -> cloudflare-1111 (1.1.1.1:80)
Test [all-ingress-deny]:
  ❌ all-ingress-deny/pod-to-cidr/cloudflare-1111-0: cilium-test/client2-6ff4656b7d-wjj8j (10.2[44](https://github.com/cilium/cilium/actions/runs/3468759658/jobs/5794941960#step:12:45).1.237) -> cloudflare-1111 (1.1.1.1:80)
  ❌ all-ingress-deny/pod-to-cidr/cloudflare-1111-1: cilium-test/client-7bdbddd7b-bc67l (10.244.1.97) -> cloudflare-1111 (1.1.1.1:80)
Test [to-cidr-1111]:
  ❌ to-cidr-1111/pod-to-cidr/cloudflare-1111-0: cilium-test/client-7bdbddd7b-bc67l (10.244.1.97) -> cloudflare-1111 (1.1.1.1:80)
  ❌ to-cidr-1111/pod-to-cidr/cloudflare-1111-1: cilium-test/client2-6ff[46](https://github.com/cilium/cilium/actions/runs/3468759658/jobs/5794941960#step:12:47)56b7d-wjj8j (10.244.1.[237](https://github.com/cilium/cilium/actions/runs/3468759658/jobs/5794941960#step:12:238)) -> cloudflare-1111 (1.1.1.1:80)
Test [client-egress-to-cidr-deny]:
  ❌ client-egress-to-cidr-deny/pod-to-cidr/cloudflare-1111-0: cilium-test/client-7bdbddd7b-bc67l (10.[244](https://github.com/cilium/cilium/actions/runs/3468759658/jobs/5794941960#step:12:245).1.97) -> cloudflare-1111 (1.1.1.1:80)
  ❌ client-egress-to-cidr-deny/pod-to-cidr/cloudflare-1111-1: cilium-test/client2-6ff4656b7d-wjj8j (10.244.1.237) -> cloudflare-1111 (1.1.1.1:80)
Test [client-egress-l7-named-port]:
  ❌ client-egress-l7-named-port/pod-to-world/http-to-one-one-one-one-1: cilium-test/client2-6ff4656b7d-wjj8j (10.244.1.237) -> one-one-one-one-http (one.one.one.one:80)
Error: Process completed with exit code 1.
```

**Comment by joestringer** [comment]
I suspect that the same underlying cause is also triggering failures in the "
Smoke test / conformance-test" failure like the following:

```
Run kubectl apply -f examples/kubernetes/connectivity-check/connectivity-check.yaml
  kubectl apply -f examples/kubernetes/connectivity-check/connectivity-check.yaml
  kubectl wait --for=condition=Available --all deployment --timeout=[2](https://github.com/cilium/cilium/actions/runs/3468759657/jobs/5794960930#step:10:2)m
  shell: /usr/bin/bash -e {0}
  env:
    cilium_cli_version: v0.12.4
    KIND_VERSION: v0.11.1
    KIND_CONFIG: .github/kind-config.yaml
    CONFORMANCE_TEMPLATE: examples/kubernetes/connectivity-check/connectivity-check.yaml
    TIMEOUT: 2m
    LOG_TIME: [3](https://github.com/cilium/cilium/actions/runs/3468759657/jobs/5794960930#step:10:3)0m
    PROM_VERSION: 2.3[4](https://github.com/cilium/cilium/actions/runs/3468759657/jobs/5794960930#step:10:4).0
    QUAY_ORGANIZATION: cilium
    QUAY_ORGANIZATION_DEV: cilium
deployment.apps/echo-a created
deployment.apps/echo-b created
deployment.apps/echo-b-host created
deployment.apps/pod-to-a created
deployment.apps/pod-to-external-[11](https://github.com/cilium/cilium/actions/runs/3468759657/jobs/5794960930#step:10:11)11 created
deployment.apps/pod-to-a-denied-cnp created
deployment.apps/pod-to-a-allowed-cnp created
deployment.apps/pod-to-external-fqdn-allow-google-cnp created
deployment.apps/pod-to-b-multi-node-clusterip created
deployment.apps/pod-to-b-multi-node-headless created
deployment.apps/host-to-b-multi-node-clusterip created
deployment.apps/host-to-b-multi-node-headless created
deployment.apps/pod-to-b-multi-node-nodeport created
deployment.apps/pod-to-b-intra-node-nodeport created
service/echo-a created
service/echo-b created
service/echo-b-headless created
service/echo-b-host-headless created
ciliumnetworkpolicy.cilium.io/pod-to-a-denied-cnp created
ciliumnetworkpolicy.cilium.io/pod-to-a-allowed-cnp created
ciliumnetworkpolicy.cilium.io/pod-to-external-fqdn-allow-google-cnp created
deployment.apps/echo-a condition met
deployment.apps/echo-b condition met
deployment.apps/echo-b-host condition met
deployment.apps/host-to-b-multi-node-clusterip condition met
deployment.apps/host-to-b-multi-node-headless condition met
deployment.apps/pod-to-a condition met
deployment.apps/pod-to-a-allowed-cnp condition met
deployment.apps/pod-to-a-denied-cnp condition met
deployment.apps/pod-to-b-intra-node-nodeport condition met
deployment.apps/pod-to-b-multi-node-clusterip condition met
deployment.apps/pod-to-b-multi-node-headless condition met
deployment.apps/pod-to-b-multi-node-nodeport condition met
deployment.apps/pod-to-external-fqdn-allow-google-cnp condition met
error: timed out waiting for the condition on deployments/pod-to-external-1111
Error: Process completed with exit code 1.
```

Example:
https://github.com/cilium/cilium/actions/runs/3468759657/jobs/5794960930

**Comment by brb** [comment]
Checked one particular failure (https://github.com/cilium/cilium/actions/runs/3487263981/jobs/5835889748):
```
[2022-11-17T11:28:55Z]   [.] Action [no-policies/pod-to-cidr/cloudflare-1111-0: cilium-test/client-594cbbdf44-xfdth (10.244.1.45) -> cloudflare-1111 (1.1.1.1:80)]
[2022-11-17T11:28:55Z]   🐛 Executing command [curl -w %{local_ip}:%{local_port} -> %{remote_ip}:%{remote_port} = %{response_code} --silent --fail --show-error --connect-timeout 5 --output /dev/null http://1.1.1.1:80]
[2022-11-17T11:29:00Z]   ❌ command "curl -w %{local_ip}:%{local_port} -> %{remote_ip}:%{remote_port} = %{response_code} --silent --fail --show-error --connect-timeout 5 --output /dev/null [http://1.1.1.1:80](http://1.1.1.1/)" failed: command terminated with exit code 28
```

The hubble flow output:
```
{"flow":{"time":"2022-11-17T11:28:55.993583582Z","verdict":"FORWARDED","ethernet":{"source":"46:54:19:b9:f8:13","destination":"5a:f4:38:ca:fd:dd"},"IP":{"source":"10.244.1.45","destination":"1.1.1.1","ipVersion":"IPv4"},"l4":{"TCP":{"source_port":48818,"destination_port":80,"flags":{"SYN":true}}},"source":{"ID":2880,"identity":547,"namespace":"cilium-test","labels":["k8s:io.cilium.k8s.policy.cluster=kind-chart-testing","k8s:io.cilium.k8s.policy.serviceaccount=client","k8s:io.kubernetes.pod.namespace=cilium-test","k8s:kind=client","k8s:name=client"],"pod_name":"client-594cbbdf44-xfdth","workloads":[{"name":"client","kind":"Deployment"}]},"destination":{"identity":2,"labels":["reserved:world"]},"Type":"L3_L4","node_name":"kind-chart-testing/chart-testing-worker","event_type":{"type":4,"sub_type":5},"trace_observation_point":"FROM_ENDPOINT","Summary":"TCP Flags: SYN"},"node_name":"kind-chart-testing/chart-testing-worker","time":"2022-11-17T11:28:55.993583582Z"}
```
tcpdump recording on that host running the Kind cluster:
```
12:28:55.993610 IP 172.18.0.3.48818 > 1.1.1.1.80: Flags [S], seq 3622202664, win 64860, options [mss 1410,sackOK,TS val 1670132751 ecr 0,nop,wscale 7], length 0
12:28:55.993610 IP 172.18.0.3.48818 > 1.1.1.1.80: Flags [S], seq 3622202664, win 64860, options [mss 1410,sackOK,TS val 1670132751 ecr 0,nop,wscale 7], length 0
12:28:55.993626 IP 10.1.0.113.48818 > 1.1.1.1.80: Flags [S], seq 3622202664, win 64860, options [mss 1410,sackOK,TS val 1670132751 ecr 0,nop,wscale 7], length 0
```
So, the packet was sent by the host. However, no visibility where it was lost.

**Comment by joestringer** [comment]
@brb based on the hubble flow there, it looks like Cilium is delivering the packet to the network with the source IP of the pod and not the node. Do we anticipate that something later on will take care of masquerading the source of the traffic to a routable IP address? Otherwise we can deliver the packets as much as we like, but Cloudflare may not be able to route the responses back to this host.

**Comment by brb** [comment]
It gets masqueraded two times:
- `172.18.0.3.48818` to NodeIP by Cilium's iptables MASQUERADE.
- `10.1.0.113.48818` to host IP by Docker's iptables MASQUERADE.

**Comment by pchaigno** [comment]
@brlbil What's the current state or this? Are we still relying on 1.1.1.1 for that test?

**Comment by brlbil** [comment]
@pchaigno we have merged #23986 which makes use of different IP addresses for different cloud providers. 
We use 8.8.8.8, and 1.1.1.1 combinations where it works best. So in practice, we have been testing with external cidrs for a while now.
I did not want to close the issue at the time in case we got more flakes. If we have not been getting flakes maybe we can close the issue now.

**Comment by pchaigno** [comment]
There seem to be some ongoing flakes with those specific tests, but nobody is sharing details. Everyone is pointing to this to skip the CI job. Let's close it and hopefully that will encourage people to open new issues with more details.
